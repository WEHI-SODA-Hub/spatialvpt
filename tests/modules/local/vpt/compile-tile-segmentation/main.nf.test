nextflow_process {

    name "Test Process COMPILE_TILE_SEGMENTATION"
    script "modules/local/vpt/compile-tile-segmentation/main.nf"
    process "COMPILE_TILE_SEGMENTATION"

    test("Should run without failures") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = Channel.of([
                    [ id: 'small_set' ],
                    [ file('${projectDir}/tests/data/segmentation_specification.json', checkIfExists: true) ],
                    [ file('${projectDir}/data/202305010900_U2OS_small_set_VMSC00000/region_0/images', checkIfExists: true) ],
                    [ file('${projectDir}/data/example_analysis_algorithm/cellpose_default_1_ZLevel.json', checkIfExists: true) ]
                ])
                input[1] = Channel.fromPath('${projectDir}/tests/data/result_tiles/cell_*.parquet').collect()
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
