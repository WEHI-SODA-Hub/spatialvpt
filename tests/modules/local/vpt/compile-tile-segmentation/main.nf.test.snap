{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "small_set"
                        },
                        "cellpose2_mosaic_space.parquet:md5,6976ef94ae61cc909f5d86751885e47b"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "small_set"
                        },
                        "cellpose2_micron_space.parquet:md5,b2fd08c50d641efab7e0811e3cade36f"
                    ]
                ],
                "2": [
                    "versions.yml:md5,3a96416e77ec12ca664b4ef357572acc"
                ],
                "micron_space": [
                    [
                        {
                            "id": "small_set"
                        },
                        "cellpose2_micron_space.parquet:md5,b2fd08c50d641efab7e0811e3cade36f"
                    ]
                ],
                "mosaic_space": [
                    [
                        {
                            "id": "small_set"
                        },
                        "cellpose2_mosaic_space.parquet:md5,6976ef94ae61cc909f5d86751885e47b"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3a96416e77ec12ca664b4ef357572acc"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-08-09T13:06:15.001243575"
    }
}